napari/utils/events/evented_model.py

```diff
@@ -309,15 +319,33 @@ def update(
 
         with self.events.blocker() as block:
             for key, value in values.items():
-                field = getattr(self, key)
-                if isinstance(field, EventedModel) and recurse:
-                    field.update(value, recurse=recurse)
-                else:
-                    setattr(self, key, value)
+                setattr(self, key, value)
 
         if block.count:
             self.events(Event(self))
 
+    def _update_inplace(self, other):
+        self.update(other)
+
+    def _uneventful(self):
+        return self.dict()
+
+    def _validate(self, new_values):
+        """
+        validate values against the current model. This differs from pydantic's public
+        validate method because it won't return an instance of Self (expensive for evented models)
+        """
+        # use non-evented versions of object so we don't trigger anything
```

Comment:
question from below still applies here though ... would it not be possible to use a context_manager to block events here, rather than constructing a new object that lacks events?